src/Type/Generic/TemplateTypeFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Generic;
+
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+
+final class TemplateTypeFactory
+{
+
+	public static function create(TemplateTypeScope $scope, string $name, ?Type $bound): Type
```

Comment:
Does this class has any usage besides tests?

Reply:
Yes, in later PRs, this class is used by `PhpDocNodeResolver` to create the right `TemplateType` instance from the given bound.

More generally, it hides the fact that `TemplateType` is actually implemented by a different class for each bound.